pkg/workflows/workload/upgrade.go

```diff
@@ -0,0 +1,62 @@
+package workload
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/workflows/interfaces"
+)
+
+// Upgrade is a schema for upgrade cluster.
+type Upgrade struct {
+	provider         providers.Provider
+	clusterManager   interfaces.ClusterManager
+	gitOpsManager    interfaces.GitOpsManager
+	writer           filewriter.FileWriter
+	eksdInstaller    interfaces.EksdInstaller
+	ClusterUpgrader  interfaces.ClusterUpgrader
+	packageInstaller interfaces.PackageInstaller
+}
+
+// NewUpgrade builds a new upgrade construct.
+func NewUpgrade(provider providers.Provider,
+	clusterManager interfaces.ClusterManager, gitOpsManager interfaces.GitOpsManager,
+	writer filewriter.FileWriter,
+	clusterUpgrade interfaces.ClusterUpgrader,
+	eksdInstaller interfaces.EksdInstaller,
+	packageInstaller interfaces.PackageInstaller,
+) *Upgrade {
+	return &Upgrade{
+		provider:         provider,
+		clusterManager:   clusterManager,
+		gitOpsManager:    gitOpsManager,
+		writer:           writer,
+		eksdInstaller:    eksdInstaller,
+		ClusterUpgrader:  clusterUpgrade,
+		packageInstaller: packageInstaller,
+	}
+}
+
+// Run Upgrade implements upgrade functionality for workload cluster's upgrade operation.
+func (c *Upgrade) Run(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec, validator interfaces.Validator) error {
+	logger.Info("POC New workflow creating workload cluster using the controller")
```

Comment:
we already a similar log at the cmd level
if you wanna keep both just make sure we clean them up when removing the feature toggle